apluslms-yamlidator/apluslms_yamlidator/document.py

```diff
@@ -322,10 +355,49 @@ def mlset(self, keys, value):
         container, key = find_ml(self._data, keys)
         container[key] = value
 
+    def mlset_cast(self, keys, value):
+        type_msg = "The type of {} should be {}."
+
+        if isinstance(keys, str):
+            key_list = keys.split(".")
+        elif not isinstance(keys, list):
+            key_list = list(keys)
+
+        schema_keys = list(chain.from_iterable(zip(['properties'] * len(key_list), key_list)))
+
+        container, key = find_ml(self.validator.schema, schema_keys)
+
+        if key in container:
+            container = container[key]
+            if "type" in container:
+                val_type = container["type"]
+                if val_type == "integer":
+                    if value.isdigit():
+                        value = int(value)
+                    else:
+                        raise TypeError(type_msg.format(keys, val_type))
+                elif val_type == "boolean":
+                    if value.lower == "true" or value == "1":
```

Comment:
Converting a string to a boolean, could be done in utility file. ValueError is better Exception and probably more suited for that.

For comparison, the similar logic in configparser:
https://github.com/python/cpython/blob/master/Lib/configparser.py#L826
https://github.com/python/cpython/blob/master/Lib/configparser.py#L1159
https://github.com/python/cpython/blob/master/Lib/configparser.py#L598

For now, 0, 1, true and false is good. In a long run, supporting language specific values would be cool.

Reply:
in which languages are there some other options than those...?

Reply:
For Finnish "kyllä" and "ei". As this is parsing user input from command line, it could be argued that when cli is used with `LC_CTYPE=fi_FI` we should parse also those strings. This can bee looked when (and if) cli is translated.

Reply:
ah as in real languages, not programming languages :p

Reply:
in which utility file should this be? if it's added to __init__ it could be imported like `from utils import X`, right? module_resources kind of seems like a 'misc' file.

Reply:
`module_resources` contains stuff handling resources files bundled within python packages. It extens functionality provided by importlib.resources (or backported library).

You can put it in utils/__init__.py for now.